_drafts/gdb-python-api-pypi-packages.md

```diff
@@ -0,0 +1,453 @@
+7: Addr: 0x20007000, uuid: 5e20fb38-a84e-a614-9325-562444df598d
+8: Addr: 0x20006fe0, uuid: 7a96092c-a557-7464-c4af-1528a4e957db
+9: Addr: 0x20006fc8, uuid: 2dcf4629-04b4-78d8-68a7-ff3f2bf1fcd9
+```
+
+**After:**
+
+```
+(gdb) uuid_list_dump s_list_head
+Args Passed: s_list_head
+Found a Linked List with 10 nodes:
++-----+-------------+--------------------------------------+
+| Idx |     Addr    |                 UUID                 |
++-----+-------------+--------------------------------------+
+|  0  | 0x1002002d0 | 008e7897-970e-3462-07d7-5e47a158298e |
+|  1  | 0x1002002b0 | deb2f6b5-e24c-4fdd-9f88-67bdce1ff261 |
+|  2  | 0x100200290 | 818fc33a-1e0f-c02c-4455-7ac8ab50c9b2 |
+|  3  | 0x100200270 | 0ce10698-6d61-ff34-a11e-19fd3650e9b7 |
+|  4  | 0x100200250 | d3feeda5-d5f3-d9e4-5bfa-6cc351e220ae |
+|  5  | 0x100200230 | e9c5f1b0-c415-8ae5-9b4d-39f6f7e8a105 |
+|  6  | 0x100200210 | 02353981-84b8-14a2-9cb4-5a672acae548 |
+|  7  | 0x1002001f0 | 48336241-f30d-23e5-5f30-d1c8ed610c4b |
+|  8  | 0x100200080 | 17119854-2f11-2d05-58f5-6bd688079992 |
+|  9  | 0x100200060 | a7f1d92a-82c8-d8fe-434d-98558ce2b347 |
++-----+-------------+--------------------------------------+
+```
+
+This is incredibly easy to do using the PTable package, as using the package is
+as simple as:
+
+```
+x = PrettyTable()
+x.field_names = ["Field Name 1", ...]
+x.add_row(["Value 1", ...])
+...
+print(x)
+```
+
+Lets go ahead and try it out!
+
+_Like Interrupt? [Subscribe](http://eepurl.com/gpRedv) to get our latest
+posts straight to your mailbox_
+
+## Setting up GDB's Python and PyPi
+
+If you want to know the solution right now and skip over our investigation step,
+continue from [here](#setting-syspath-within-gdbinit).
+
+### Initial Notes
+
+Importing PyPi packages within your current installation might work already, but
+that is likely because the installed GDB was compiled locally and against a
+locally installed Python version.
+
+If you installed GDB from standard package managers such as `brew` on macOS,
+`apt` on Ubuntu, or downloaded it from official sources such as
+[gnu.org](https://www.gnu.org/software/gdb/download/) or
+[ARM](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads),
+then the following steps are likely required. These binaries are built against
+the system Python installation and will not automatically source packages from a
+virtual environment.
+
+Regardless of whether GBD currently imports PyPi packages correctly now due to
+the installation method, the steps in this tutorial are still heavily suggested,
+as it will allow GDB to always read the packages from the currently activated
+virtual environment in the users shell.
+
+Throughout this tutorial, I will be using:
+
+- macOS 10.14 using Python 2.7.15 installed using [Brew](https://brew.sh/) along
+  with a Virtual Environment
+- `arm-none-eabi-gdb-py` which is GDB 8.2 compiled against the macOS System
+  Python 2.7.10.
+
```

Comment:
Perhaps add a paragraph on how to setup a virtualenv and why they are good (or link to an article about them).

Reply:
Still need to do this.